python/src/fast.cpp

```diff
@@ -82,23 +82,25 @@ void init_fast(nb::module_& parent_module) {
       "base"_a,
       "scale"_a,
       "offset"_a,
+      "freqs"_a = nb::none(),
       "stream"_a = nb::none(),
       nb::sig(
-          "def rope(a: array, dims: int, *, traditional: bool, base: float, scale: float, offset: int, stream: Union[None, Stream, Device] = None) -> array"),
+          "def rope(a: array, dims: int, *, traditional: bool, base: float, scale: float, offset: int, freqs: Optional[array] = None, stream: Union[None, Stream, Device] = None) -> array"),
       R"pbdoc(
         Apply rotary positional encoding to the input.
 
         Args:
             a (array): Input array.
             dims (int): The feature dimensions to be rotated. If the input feature
-                is larger than dims then the rest is left unchanged.
+              is larger than dims then the rest is left unchanged.
             traditional (bool): If set to ``True`` choose the traditional
-                implementation which rotates consecutive dimensions.
+              implementation which rotates consecutive dimensions.
             base (float): The base used to compute angular frequency for
```

Comment:
I'm thinking about making `base` an `Optional[float]` and making it an error to set both `base` and `freqs`. Wdyt?

Reply:
Yeah I like it! The less ambiguity the better!